application/services/Participants.php

```diff
@@ -306,6 +306,77 @@ public function getUnEnrolledCountriesByShipmentId($shipmentId) {
         return $countries;
     }
 
+
+     public function getEnrolledAndUnErolledParticipants($shipmentId)
+    {
+        $dbAdapter = Zend_Db_Table_Abstract::getDefaultAdapter();
+        $participantCountries = $this->getParticipantsAgainstCountries($dbAdapter);
+        $shipmentParticipantMapList = $this->getAllShipmentMapByShipmentId($dbAdapter,$shipmentId);
```

Comment:
These two calls but be faster if they were just part of the same SQL query, you could also determine if the participant was enrolled in the same SQL query.

Reply:
Hi @brichardssa we had tried a lot to optimize the query but no way we found, if you have then let me know please. I am sure these two query run once.